napari/add_layers.py

```diff
@@ -0,0 +1,633 @@
+    scale : tuple of float
+        Scale factors for the layer.
+    translate : tuple of float
+        Translation values for the layer.
+    opacity : float
+        Opacity of the layer visual, between 0.0 and 1.0.
+    blending : str
+        One of a list of preset blending modes that determines how RGB and
+        alpha values of the layer visual get mixed. Allowed values are
+        {'opaque', 'translucent', and 'additive'}.
+    visible : bool
+        Whether the layer visual is currently being displayed.
+
+    Returns
+    -------
+    viewer : :class:`napari.Viewer`
+        The newly-created viewer.
+    """
+    viewer = Viewer()
+    viewer.add_surface(
+        data,
+        colormap=colormap,
+        contrast_limits=contrast_limits,
+        name=name,
+        metadata=metadata,
+        scale=scale,
+        translate=translate,
+        opacity=opacity,
+        blending=blending,
+        visible=visible,
+    )
+    return viewer
+
+
+def view_vectors(
+    data,
+    *,
+    edge_width=1,
+    edge_color='red',
+    length=1,
+    name=None,
+    metadata=None,
+    scale=None,
+    translate=None,
+    opacity=0.7,
+    blending='translucent',
+    visible=True,
+):
+    """Create a viewer and add a vectors layer.
+
+    Parameters
+    ----------
+    data : (N, 2, D) or (N1, N2, ..., ND, D) array
+        An (N, 2, D) array is interpreted as "coordinate-like" data and a
+        list of N vectors with start point and projections of the vector in
+        D dimensions. An (N1, N2, ..., ND, D) array is interpreted as
+        "image-like" data where there is a length D vector of the
+        projections at each pixel.
+    edge_width : float
+        Width for all vectors in pixels.
+    length : float
+         Multiplicative factor on projections for length of all vectors.
+    edge_color : str
+        Edge color of all the vectors.
+    name : str
+        Name of the layer.
+    metadata : dict
+        Layer metadata.
+    scale : tuple of float
+        Scale factors for the layer.
+    translate : tuple of float
+        Translation values for the layer.
+    opacity : float
+        Opacity of the layer visual, between 0.0 and 1.0.
+    blending : str
+        One of a list of preset blending modes that determines how RGB and
+        alpha values of the layer visual get mixed. Allowed values are
+        {'opaque', 'translucent', and 'additive'}.
+    visible : bool
+        Whether the layer visual is currently being displayed.
+
+    Returns
+    -------
+    viewer : :class:`napari.Viewer`
+        The newly-created viewer.
+    """
+    viewer = Viewer()
+    viewer.add_vectors(
+        data,
+        edge_width=edge_width,
+        edge_color=edge_color,
+        length=length,
+        name=name,
+        metadata=metadata,
+        scale=scale,
+        translate=translate,
+        opacity=opacity,
+        blending=blending,
+        visible=visible,
+    )
+    return viewer
```

Comment:
Consider this comment entirely optional, just wanted to write it down somewhere.

Depending on how tempted you are by extra work that makes the code more elegant (but it is extra work), it occurs to me that this whole file could be replaced by something like:

```python
import functools
from .viewer import Viewer


def view_layer(layer_method):
    def view_func(*args, **kwargs):
        v = Viewer()
        getattr(v, layer_method.__name__)(*args, **kwargs)
        return v
    view_func.__doc__ = fix_docstring(layer_method.__doc__)
    return view_func

view_image = view_layer(Viewer.add_image)
view_pyramid = view_layer(Viewer.add_pyramid)
...
```

Of course, you'd need to write `fix_docstring`, and I think then there's no way around `*args` and `**kwargs`, which is annoying. (Perhaps the signature can be altered the same way that the doc is altered, not sure.) For fix docstring, you probably want to use numpydoc, which I've been playing with for the plugins PR. It's pretty nifty!

```
In [13]: numpydoc.get_doc_object(feature.hessian_matrix)                                         
Out[13]: <numpydoc.docscrape_sphinx.SphinxFunctionDoc at 0x7f6593757908>

In [14]: _13.get_func                                                                            
Out[14]: <bound method FunctionDoc.get_func of <numpydoc.docscrape_sphinx.SphinxFunctionDoc object at 0x7f6593757908>>

In [15]: _13.get_func()                                                                          
Out[15]: 
(<function skimage.feature.corner.hessian_matrix(image, sigma=1, mode='constant', cval=0, order='rc')>,
 'hessian_matrix')

In [16]: _13.xref_param_type                                                                     
Out[16]: False

In [17]: _13.items()                                                                             
Out[17]: ItemsView(<numpydoc.docscrape_sphinx.SphinxFunctionDoc object at 0x7f6593757908>)

In [18]: list(_13.items())                                                                       
Out[18]: 
[('Signature',
  "hessian_matrix(image, sigma=1, mode='constant', cval=0, order='rc')"),
 ('Summary', ['Compute Hessian matrix.']),
 ('Extended Summary',
  ['The Hessian matrix is defined as::',
   '',
   '    H = [Hrr Hrc]',
   '        [Hrc Hcc]',
   '',
   'which is computed by convolving the image with the second derivatives',
   'of the Gaussian kernel in the respective r- and c-directions.']),
 ('Parameters',
  [Parameter(name='image', type='ndarray', desc=['Input image.']),
   Parameter(name='sigma', type='float', desc=['Standard deviation used for the Gaussian kernel, which is used as', 'weighting function for the auto-correlation matrix.']),
   Parameter(name='mode', type="{'constant', 'reflect', 'wrap', 'nearest', 'mirror'}, optional", desc=['How to handle values outside the image borders.']),
   Parameter(name='cval', type='float, optional', desc=["Used in conjunction with mode 'constant', the value outside", 'the image boundaries.']),
   Parameter(name='order', type="{'rc', 'xy'}, optional", desc=['This parameter allows for the use of reverse or forward order of', "the image axes in gradient computation. 'rc' indicates the use of", "the first axis initially (Hrr, Hrc, Hcc), whilst 'xy' indicates the", 'usage of the last axis initially (Hxx, Hxy, Hyy)'])]),
 ('Returns',
  [Parameter(name='Hrr', type='ndarray', desc=['Element of the Hessian matrix for each pixel in the input image.']),
   Parameter(name='Hrc', type='ndarray', desc=['Element of the Hessian matrix for each pixel in the input image.']),
   Parameter(name='Hcc', type='ndarray', desc=['Element of the Hessian matrix for each pixel in the input image.'])]),
 ('Yields', []),
 ('Receives', []),
 ('Raises', []),
 ('Warns', []),
 ('Other Parameters', []),
 ('Attributes', []),
 ('Methods', []),
 ('See Also', []),
 ('Notes', []),
 ('Warnings', []),
 ('References', ''),
 ('Examples',
  ['>>> from skimage.feature import hessian_matrix',
   '>>> square = np.zeros((5, 5))',
   '>>> square[2, 2] = 4',
   ">>> Hrr, Hrc, Hcc = hessian_matrix(square, sigma=0.1, order='rc')",
   '>>> Hrc',
   'array([[ 0.,  0.,  0.,  0.,  0.],',
   '       [ 0.,  1.,  0., -1.,  0.],',
   '       [ 0.,  0.,  0.,  0.,  0.],',
   '       [ 0., -1.,  0.,  1.,  0.],',
   '       [ 0.,  0.,  0.,  0.,  0.]])']),
 ('index', {})]

In [19]: _13['Parameters']                                                                       
Out[19]: 
[Parameter(name='image', type='ndarray', desc=['Input image.']),
 Parameter(name='sigma', type='float', desc=['Standard deviation used for the Gaussian kernel, which is used as', 'weighting function for the auto-correlation matrix.']),
 Parameter(name='mode', type="{'constant', 'reflect', 'wrap', 'nearest', 'mirror'}, optional", desc=['How to handle values outside the image borders.']),
 Parameter(name='cval', type='float, optional', desc=["Used in conjunction with mode 'constant', the value outside", 'the image boundaries.']),
 Parameter(name='order', type="{'rc', 'xy'}, optional", desc=['This parameter allows for the use of reverse or forward order of', "the image axes in gradient computation. 'rc' indicates the use of", "the first axis initially (Hrr, Hrc, Hcc), whilst 'xy' indicates the", 'usage of the last axis initially (Hxx, Hxy, Hyy)'])]
```

Reply:
sounds like a great follow-up PR ;-)